common/src/test/java/tachyon/security/PlainServerCallbackHandlerTest.java

```diff
@@ -0,0 +1,88 @@
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.security;
+
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.callback.NameCallback;
+import javax.security.auth.callback.PasswordCallback;
+import javax.security.sasl.AuthenticationException;
+import javax.security.sasl.AuthorizeCallback;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+import tachyon.security.authentication.AuthenticationFactory.AuthTypes;
+import tachyon.security.authentication.AuthenticationProvider;
+
+public class PlainServerCallbackHandlerTest {
+  private TachyonConf mConf = new TachyonConf();
+  private CallbackHandler mPlainServerCBHandler =
+      new PlainSaslServer.PlainServerCallbackHandler(AuthTypes.CUSTOM, mConf);
+
+  @Rule
+  public ExpectedException mThrown = ExpectedException.none();
+
+  @Before
+  public void before() throws Exception {
+    mConf.set(Constants.TACHYON_AUTHENTICATION_PROVIDER_CUSTOM_CLASS,
+        NameMatchAuthenticationProvider.class.getName());
+  }
+
+  @Test
+  public void authenticateNameMatchTest() throws Exception {
+    String authenticateId = "tachyon-1";
+    NameCallback ncb = new NameCallback(" authentication id: ");
+    ncb.setName(authenticateId);
+
+    PasswordCallback pcb = new PasswordCallback(" password: ", false);
+    pcb.setPassword("anonymous".toCharArray());
+
+    Callback[] callbacks = new Callback[]{ncb, pcb,
+        new AuthorizeCallback(authenticateId, authenticateId)};
+    mPlainServerCBHandler.handle(callbacks);
+  }
+
+  @Test
+  public void authenticateNameNotMatchTest() throws Exception {
+    mThrown.expect(AuthenticationException.class);
+    mThrown.expectMessage("Only allow the user startWith tachyon");
+
+    String authenticateId = "not-tachyon-1";
+    NameCallback ncb = new NameCallback(" authentication id: ");
+    ncb.setName(authenticateId);
+
+    PasswordCallback pcb = new PasswordCallback(" password: ", false);
+    pcb.setPassword("anonymous".toCharArray());
+
+    Callback[] callbacks = new Callback[]{ncb, pcb,
+        new AuthorizeCallback(authenticateId, authenticateId)};
+    mPlainServerCBHandler.handle(callbacks);
+  }
+
+  public static class NameMatchAuthenticationProvider implements AuthenticationProvider {
```

Comment:
Not sure if we want to leave this class `public`---in general, helper functions/classes in unit tests are not supposed to be used externally. If you make it `private` and it still works, I think `private` is preferred then; otherwise, leave it as is.